### ISSUE TITLE
How to get fine optical flow

### ISSUE THREAD

**Issue by JamesYang-7** [vccimaging/Reconstrution_Through_Moving_Water#5]
I tried generating optical flow of example 1 using RAFT and its pretrained weights but the result was not good. Did you use pretrained model directly for optical flow or finetuned on specific date set?

**Comment by jhxiong** [comment]
I used PWC-net to compute the optical flow. The flow model was pre-trained, I found it perform reasonably well on the water fluctuation distortions. I did not try with RAFT but I suppose it will not be worse. While you compute the flow on your own, you need to pay attention to:
- make sure the flow computation to the reference frame was correct. How: warp the frame to the reference frame with the computed flows and the warped frame should match with the reference one with only few pixels exceptions.
- You need to update the confidence mask based on the above warp testing, or using all ones as a first test. Notice the confidence mask provided in the folder is for the flow computed with PWC-net, different flow model will result in different confidence mask.
